knora/dsplib/utils/shared_methods.py

```diff
@@ -0,0 +1,99 @@
+import re
+import time
+from datetime import datetime
+from typing import Union, Callable, Any, Optional
+
+from requests import RequestException
+
+from knora.dsplib.models.connection import Connection
+from knora.dsplib.models.helpers import BaseError
+
+
+def login(server: str, user: str, password: str) -> Connection:
+    """
+    Logs in and returns the active connection. Raises a BaseError if the login fails.
+
+    Args:
+        server: URL of the DSP server to connect to
+        user: Username (e-mail)
+        password: Password of the user
+
+    Return:
+        Connection instance
+    """
+    con = Connection(server)
+    try_network_action(
+        object=con,
+        method="login",
+        kwargs={"email": user, "password": password},
+        error_message_on_failure="ERROR: Cannot login to DSP server"
+    )
+    return con
+
+
+def try_network_action(
+    error_message_on_failure: str,
+    method: Union[str, Callable[..., Any]],
+    object: Optional[Any] = None,
+    kwargs: Optional[dict[str, Any]] = None
```

Comment:
as mentioned above, I think this function would be much neater if it only accepted a callable and a message (which I would maybe call `failure_msg` to make function calls a bit slimmer); and in those "special" case where you need the object or kwargs, I'd solve that with lambdas in the function call.